content/blog/2019-11-05-tidync.md

```diff
@@ -0,0 +1,1099 @@
+## 1     5 anom  NC_SHORT     4     6
+```
+
+
+Similarly for dimensions and attributes, but attributes for a variable can only be found by name. 
+
+
+```r
+ncmeta::nc_dim(oisstfile, "lon")
+```
+
+```
+## # A tibble: 1 x 4
+##      id name  length unlim
+##   <int> <chr>  <dbl> <lgl>
+## 1     0 lon      180 FALSE
+```
+
+```r
+ncmeta::nc_dim(oisstfile, 0)
+```
+
+```
+## # A tibble: 1 x 4
+##      id name  length unlim
+##   <int> <chr>  <dbl> <lgl>
+## 1     0 lon      180 FALSE
+```
+
+```r
+ncmeta::nc_atts(oisstfile)
+```
+
+```
+## # A tibble: 50 x 4
+##       id name          variable value    
+##    <int> <chr>         <chr>    <list>   
+##  1     0 standard_name lon      <chr [1]>
+##  2     1 long_name     lon      <chr [1]>
+##  3     2 units         lon      <chr [1]>
+##  4     3 axis          lon      <chr [1]>
+##  5     0 standard_name lat      <chr [1]>
+##  6     1 long_name     lat      <chr [1]>
+##  7     2 units         lat      <chr [1]>
+##  8     3 axis          lat      <chr [1]>
+##  9     0 long_name     zlev     <chr [1]>
+## 10     1 units         zlev     <chr [1]>
+## # … with 40 more rows
+```
+
+```r
+ncmeta::nc_atts(oisstfile, "zlev")
+```
+
+```
+## # A tibble: 4 x 4
+##      id name         variable value    
+##   <int> <chr>        <chr>    <list>   
+## 1     0 long_name    zlev     <chr [1]>
+## 2     1 units        zlev     <chr [1]>
+## 3     2 axis         zlev     <chr [1]>
+## 4     3 actual_range zlev     <chr [1]>
+```
+
+And we can find the internal metadata for each variable by expanding the value. 
+
+
+```r
+ncmeta::nc_atts(oisstfile, "time") %>% tidyr::unnest(cols = c(value))
+```
+
+```
+## # A tibble: 5 x 4
+##      id name          variable value                         
+##   <int> <chr>         <chr>    <chr>                         
+## 1     0 standard_name time     time                          
+## 2     1 long_name     time     Center time of the day        
+## 3     2 units         time     days since 1978-01-01 00:00:00
+## 4     3 calendar      time     standard                      
+## 5     4 axis          time     T
+```
+
+With this information we may now apply the right interpretation to the time values. In the print of the tidync object above we see the value `1460`, which is given without context in the dimensions table. 
+
+We can get that value by activating the right grid and extracting, the `time` is
+a single integer value. 
+
+
+```r
+oisst <- tidync(oisstfile)
+time_ex <- oisst %>% activate("D3") %>% hyper_array()
+time_ex$time
+```
+
+```
+## [1] 1460
+```
+
+As mentioned, tidync considers time and its metadata a bit dangerous - it is often wrong, often inconsistent, there are time zone issues, sometimes *extra seconds* to account for and so we prefer to leave these interpretations to be validated manually  before automation. 
+
+Obtain the time units information and then use it to conver the raw value (`1460`) into a date-time understood by R. 
```

Comment:
convert

Reply:
@stefaniebutland 

> 1. Could you add a couple of sentences at the top to outline what you cover?

Considering this rewrite, should I include links to the main headers? 

```
In May 2019 version 0.2.0 of [tidync](https://docs.ropensci.org/tidync/) was approved by rOpenSci and accepted to CRAN. Here we provide a quick overview of the typical workflow with some pseudo-code for the main functions in tidync, this is enough to read if you just want to try out the package on your own data.  The tidync package is focussed on *efficient data extraction* for developing your own software, and this somewhat long post takes the time to explain the concepts in detail.  

There is a section about the [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) data model itself. Then there is  a detailed illustration of a raster data set in R including some of challenges faced by R users, followed by sections on how tidync sees coordinates in NetCDF, and how we can *slice a dataset* and control the format of the output. We then discuss some limitations and future work, and then (most importantly) reflect on the rOpenSci process of package review. 
```

Reply:
done